### PR TITLE
SUBMARINE-616. Deploying Traefik as Ingress controller in Submarine 

### PR DESCRIPTION
**Pull request by lowc1012**
### What is this PR for?
1. Declaring traefik-helm-chart as a dependency(subchart). And  it should be optional.
2. Add an Ingress object to route to submarine-server service and using NodePort (port 32080/TCP) to expose.

After installing Submarine with helm, users can directly access submarine-workbench from your browser.
If users dont want to use Traefik as default ingress controller, they can modify the value ".Value.submarine.traefik.enabled"

### What type of PR is it?
[Feature]

### Todos
* [ ] - Task

### What is the Jira issue?
[SUBMARINE-616](https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-616)

### How should this be tested?
[Travis CI](https://travis-ci.org/github/lowc1012/submarine/builds/724193208)

### Screenshots (if appropriate)
image1

image2

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -40,10 +40,34 @@ jobs:
     timeout-minutes: 120
     steps:
       - uses: actions/checkout@v1
+      - name: Create the kind config
+        run: |
+          configFile=/home/runner/work/submarine/submarine/kind-config-kind.yaml
```
Comment (liuxunorg) [inline .github/workflows/python.yml:45]: `/home/runner/work/submarine/submarine/kind-config-kind.yaml` is this path fixed?

**Review by liuxunorg — CHANGES_REQUESTED** [review]
For newly added files, the file name needs to be modified.
For example: `clusterrolebinding.yaml` needs to be modified to `cluster-role-binding.yaml`, `clusterrole.yaml` is modified to `cluster-role.yaml`, etc.

File: .github/workflows/python.yml (hunk 1 above)
Comment (lowc1012) [reply .github/workflows/python.yml:45]: If we set config: ./kind-config-kind.yaml (line: 70), KinD will set the path to a config file under "/home/runner/work/submarine/submarine/".
I can replace the configFile variable to "./kind-config-kind.yaml" (just replace it with the relative path)

**Comment by lowc1012** [comment]
There seems to be something bug. WIP

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -40,10 +40,33 @@ jobs:
     timeout-minutes: 120
     steps:
       - uses: actions/checkout@v1
+      - name: Create the kind config
+        run: |
+          cat <<EOF > ./kind-config-kind.yaml
```
[the remark this answers is not in the document]
Comment (lowc1012) [reply .github/workflows/python.yml:45]: Either will be fine. kind-config or kinf-config-{cluster name}

File: docs/userdocs/k8s/helm.md [hunk 3]
```diff
@@ -72,15 +74,36 @@ notebook-controller-deployment-5db8b6cbf7-k65jm   1/1     Running   0          5
...
+use Traefik. You can modify below value in `./helm-charts/submarine/values.yaml`.
+```yaml
+# Use Traefik by default
+traefik:
+  enabled: true
+```
+
+To access the server from outside of the cluster, we use Traefik ingress controller and
+NodePort for external access.\
+Please refer to `./helm-charts/submarine/charts/traefik/values.yaml` and [Traefik docs}(https://docs.traefik.io/)
```
[the remark this answers is not in the document]
Comment (lowc1012) [reply docs/userdocs/k8s/helm.md]: done

File: docs/userdocs/k8s/helm.md [hunk 4]
```diff
@@ -72,15 +74,36 @@ notebook-controller-deployment-5db8b6cbf7-k65jm   1/1     Running   0          5
...
 ```
 
-### Enable local access to Submarine Server
-Submarine server by default expose 8080 port within K8s cluster.
-To access the server from outside of the cluster, we need to expose the service.
-We can either use port-forward, or use K8s `Ingress`, here is an example of port-forward.
+### Access to Submarine Server
+Submarine server by default expose 8080 port within K8s cluster. After Submarine v0.5
+uses Traefik as reverse-proxy by default. If you don't want to
+use Traefik. You can modify below value in `./helm-charts/submarine/values.yaml`.
```
[the remark this answers is not in the document]
Comment (lowc1012) [reply docs/userdocs/k8s/helm.md]: done

**Comment by liuxunorg** [comment]
@lowc1012 Can you show the running effect of Traefik through screenshots?

**Comment by lowc1012** [comment]
@liuxunorg 
Ok, I updated the screenshots section.

**Review by pingsutw — APPROVED** [review]
LGTM

**Pull request closed without merging**